File: objectiveai-sdk-js/src/auth/http.ts
import { ObjectiveAI, type RequestOptions } from "../client";
import type { AuthCreateApiKeyRequest } from "./createApiKeyRequest";
import type { AuthApiKeyWithMetadata } from "./apiKeyWithMetadata";
import type { AuthDisableApiKeyRequest } from "./disableApiKeyRequest";
import type { AuthListApiKeyResponse } from "./listApiKeyResponse";
import type { AuthGetCreditsResponse } from "./getCreditsResponse";
import type { AuthCreateOpenRouterByokApiKeyRequest } from "./createOpenRouterByokApiKeyRequest";
import type { AuthGetOpenRouterByokApiKeyResponse } from "./getOpenRouterByokApiKeyResponse";

export function authCreateApiKey(
  client: ObjectiveAI,
  body: AuthCreateApiKeyRequest,
  options?: RequestOptions,
): Promise<AuthApiKeyWithMetadata> {
  return client.post_unary<AuthApiKeyWithMetadata>("/auth/keys", body, options);
}

export function authCreateOpenrouterByokApiKey(
  client: ObjectiveAI,
  body: AuthCreateOpenRouterByokApiKeyRequest,
  options?: RequestOptions,
): Promise<AuthGetOpenRouterByokApiKeyResponse> {
  return client.post_unary<AuthGetOpenRouterByokApiKeyResponse>(
    "/auth/keys/openrouter",
    body,
    options,
  );
}

export function authDisableApiKey(
  client: ObjectiveAI,
  body: AuthDisableApiKeyRequest,
  options?: RequestOptions,
): Promise<AuthApiKeyWithMetadata> {
  return client.delete_unary<AuthApiKeyWithMetadata>("/auth/keys", body, options);
}

export function authDeleteOpenrouterByokApiKey(
  client: ObjectiveAI,
  options?: RequestOptions,
): Promise<void> {
  return client.delete_unary<void>("/auth/keys/openrouter", undefined, options);
}

export function authListApiKeys(
  client: ObjectiveAI,
  options?: RequestOptions,
): Promise<AuthListApiKeyResponse> {
  return client.get_unary<AuthListApiKeyResponse>("/auth/keys", options);
}

export function authGetOpenrouterByokApiKey(
  client: ObjectiveAI,
  options?: RequestOptions,
): Promise<AuthGetOpenRouterByokApiKeyResponse> {
  return client.get_unary<AuthGetOpenRouterByokApiKeyResponse>(
    "/auth/keys/openrouter",
    options,
  );
}

export function authGetCredits(
  client: ObjectiveAI,
  options?: RequestOptions,
): Promise<AuthGetCreditsResponse> {
  return client.get_unary<AuthGetCreditsResponse>("/auth/credits", options);
}

File: objectiveai-sdk-py/objectiveai_sdk/auth/http.py
"""HTTP functions for authentication endpoints."""

from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from objectiveai_sdk.auth import (
        ApiKeyWithMetadata,
        CreateApiKeyRequest,
        CreateOpenRouterByokApiKeyRequest,
        DisableApiKeyRequest,
        GetCreditsResponse,
        GetOpenRouterByokApiKeyResponse,
        ListApiKeyResponse,
    )
    from objectiveai_sdk.client import ObjectiveAI


async def create_api_key(
    client: ObjectiveAI, body: CreateApiKeyRequest,
) -> ApiKeyWithMetadata:
    """Create a new API key."""
    return await client.post_unary("auth/keys", body)


async def create_openrouter_byok_api_key(
    client: ObjectiveAI, body: CreateOpenRouterByokApiKeyRequest,
) -> GetOpenRouterByokApiKeyResponse:
    """Create or update the OpenRouter BYOK API key."""
    return await client.post_unary("auth/keys/openrouter", body)


async def disable_api_key(
    client: ObjectiveAI, body: DisableApiKeyRequest,
) -> ApiKeyWithMetadata:
    """Disable an existing API key."""
    return await client.delete_unary("auth/keys", body)


async def delete_openrouter_byok_api_key(client: ObjectiveAI) -> None:
    """Delete the OpenRouter BYOK API key."""
    return await client.delete_unary("auth/keys/openrouter")


async def list_api_keys(client: ObjectiveAI) -> ListApiKeyResponse:
    """List all API keys for the authenticated user."""
    return await client.get_unary("auth/keys")


async def get_openrouter_byok_api_key(
    client: ObjectiveAI,
) -> GetOpenRouterByokApiKeyResponse:
    """Retrieve the configured OpenRouter BYOK API key."""
    return await client.get_unary("auth/keys/openrouter")


async def get_credits(client: ObjectiveAI) -> GetCreditsResponse:
    """Retrieve the user's credit balance."""
    return await client.get_unary("auth/credits")
